react-components/src/components/RuleBasedOutputs/RuleBasedOutputsSelector.tsx

```diff
@@ -43,45 +66,65 @@ export function RuleBasedOutputsSelector({
 
   useEffect(() => {
     if (assetMappings === undefined || models === undefined || isFetching) return;
+    if (timeseriesExternalIdsFromRule.length > 0 && isLoadingAssetIdsAndTimeseriesData) return;
 
     setStylingsGroups(EMPTY_ARRAY);
 
     if (ruleSet === undefined) return;
 
-    const initializeRuleBasedOutputs = async (model: CogniteCadModel): Promise<void> => {
-      // parse assets and mappings
-      // TODO: refactor to be sure to filter only the mappings/assets for the current model within the pages
-      const flatAssetsMappingsList = assetMappings.pages
-        .flat()
-        .map((item) => item.mappings)
-        .flat();
-      const flatMappings = flatAssetsMappingsList.map((node) => node.items).flat();
-      const contextualizedAssetNodes = assetMappings.pages
-        .flat()
-        .flatMap((item) => item.assets)
-        .map(convertAssetMetadataKeysToLowerCase);
-
-      const collectionStylings = await generateRuleBasedOutputs(
-        model,
-        contextualizedAssetNodes,
-        flatMappings,
-        ruleSet
-      );
-
-      setStylingsGroups(collectionStylings);
-    };
-
     models.forEach(async (model) => {
       if (!(model instanceof CogniteCadModel)) {
         return;
       }
-      await initializeRuleBasedOutputs(model);
+      setStylingsGroups(
+        await initializeRuleBasedOutputs({
+          model,
+          assetMappings,
+          contextualizedAssetNodes,
+          ruleSet,
+          assetIdsAndTimeseries: assetIdsWithTimeseriesData?.assetIdsWithTimeseries ?? [],
+          timeseriesDatapoints: assetIdsWithTimeseriesData?.timeseriesDatapoints ?? []
+        })
+      );
     });
-  }, [assetMappings, ruleSet]);
+  }, [isLoadingAssetIdsAndTimeseriesData, ruleSet]);
 
   return <></>;
 }
 
+const initializeRuleBasedOutputs = async ({
```

Comment:
Is there a reason this is a lambda-function and not a normal function?

Reply:
ops.. no. Changed here:
https://github.com/cognitedata/reveal/pull/4414/commits/82528c8036f90d3a908a0dc63f1693254f5455bd